### PR TITLE
build: update deps version according to issue

### PR DESCRIPTION
**Pull request by wenytang-ms**
https://github.com/OfficeDev/TeamsFx-Samples/issues/1173

### CODE REVIEW

**Review by tecton — DISMISSED** [review]

**Review by KennethBWSong — APPROVED** [review]

**Review by formulahendry — DISMISSED** [review]

File: hello-world-tab-with-backend/api/package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "test": "echo \"Error: no test specified\" && exit 1"
   },
   "dependencies": {
-    "@azure/functions": "^1.2.2",
```
Comment (huimiu) [inline hello-world-tab-with-backend/api/package.json:17]: Not just about this version, there are other changes required for the Azure Functions V4 upgrade. Here is a [doc ](https://microsoftapc.sharepoint.com/:w:/t/DevDivTeamsDevXProductTeam/Ee59AXq1soBHkQtriEG2rZYBpSJYYJQh7kJ4ZQai9Z18AQ?e=BqbSXK)you can take a look.

File: todo-list-with-Azure-backend-M365/api/package.json [hunk 2]
```diff
@@ -9,10 +9,10 @@
     "test": "echo \"Error: no test specified\" && exit 1"
   },
   "dependencies": {
-    "@azure/functions": "^1.2.2",
```
Comment (swatDong) [inline todo-list-with-Azure-backend-M365/api/package.json:12]: + @huimiu for help - is it safe to just upgrade this dependency without any runtime related change?

**Review by JerryYangKai — DISMISSED** [review]

File: todo-list-with-Azure-backend-M365/api/package.json (hunk 2 above)
Comment (huimiu) [reply todo-list-with-Azure-backend-M365/api/package.json:12]: Yes, there are other things that need to be modified as well. Synced with Wenyu。

File: hello-world-tab-with-backend/api/package.json (hunk 1 above)
Comment (wenytang-ms) [reply hello-world-tab-with-backend/api/package.json]: revert function's change will send ticket to owner

File: todo-list-with-Azure-backend-M365/api/package.json (hunk 2 above)
Comment (wenytang-ms) [reply todo-list-with-Azure-backend-M365/api/package.json:12]: revert

**Review by kuojianlu — APPROVED** [review]

**Review by swatDong — DISMISSED** [review]

**Review by huimiu — DISMISSED** [review]

**Review by hund030 — APPROVED** [review]

File: todo-list-SPFx/src/package.json [hunk 3]
```diff
@@ -13,16 +13,16 @@
...
     "@microsoft/sp-adaptive-card-extension-base": "1.18.2",
     "@microsoft/sp-core-library": "1.18.2",
     "@microsoft/sp-lodash-subset": "1.18.2",
     "@microsoft/sp-office-ui-fabric-core": "1.18.2",
     "@microsoft/sp-property-pane": "1.18.2",
     "@microsoft/sp-webpart-base": "1.18.2",
-    "react": "17.0.1",
-    "react-dom": "17.0.1",
-    "tslib": "2.3.1"
+    "react": "18.2.0",
```
Comment (yuqizhou77) [inline todo-list-SPFx/src/package.json:23]: I think upgrading to react 18 will cause issues in SPFx samples. 
image

File: react-retail-dashboard/src/package.json [hunk 4]
```diff
@@ -24,10 +24,10 @@
     "@microsoft/sp-webpart-base": "1.18.2",
     "@pnp/common": "^2.15.0",
     "@pnp/sp": "^3.15.0",
-    "@pnp/spfx-controls-react": "3.15.0",
-    "react": "17.0.1",
-    "react-dom": "17.0.1",
-    "tslib": "2.3.1"
+    "@pnp/spfx-controls-react": "3.17.0",
+    "react": "18.2.0",
```
Comment (yuqizhou77) [inline react-retail-dashboard/src/package.json:28]: Upgrading to react 18 will cause issues in SPFx samples.
image

File: todo-list-SPFx/src/package.json (hunk 3 above)
Comment (wenytang-ms) [reply todo-list-SPFx/src/package.json]: thank you, revert

File: react-retail-dashboard/src/package.json (hunk 4 above)
Comment (wenytang-ms) [reply react-retail-dashboard/src/package.json]: thank you, revert

**Review by yuqizhou77 — APPROVED** [review]

**Review by tecton — APPROVED** [review]

**Review by xzf0587 — APPROVED** [review]

**Pull request merged**